packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx

```diff
@@ -134,15 +137,28 @@ const StakeExchangeForm = ({
       await createStakeTransaction.mutateAsync({
         id: '1',
         transaction_hash: txReceipt.transactionHash,
-        community_id: app.activeChainId(),
+        community_id: communityId,
       });
 
       onSetSuccessTransactionHash(txReceipt?.transactionHash);
       onSetModalState(ManageCommunityStakeModalState.Success);
 
+      // join user to community if not already a member
```

Comment:
what's the reason for joining the community after buying a stake, instead of joining before buying stake?

Reply:
this was a requirement from the product, which is making it easier to buy stake from global explore, and to make the behaviour consistent, we join users to community on any page they buy stake from

from https://github.com/hicommonwealth/commonwealth/issues/6594
> User story: I am looking at communities with Stake enabled, and I am acquiring a small batch of stake in every community. I dont take the time to join the communities -- i just go in and mint some on Commonwealth.im. Afterwards, I see that the community has shown up on the sidebar of the navigation system.